### PR TITLE
feat: update mac install script to download from latest release

### PR DESCRIPTION
**Pull request by obs-gh-mattcotter**
### Description

Update mac install script to download from latest release. Now that we have a new release which builds the expected Mac zip, we can finally download and install from GitHub.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: scripts/install_mac.sh [hunk 1]
```diff
@@ -40,17 +46,32 @@ sudo mkdir -p $observeagent_install_dir /usr/local/libexec /usr/local/bin /var/l
...
 sudo chown -R root:wheel $observeagent_install_dir
 
 # Initialize the agent config file if it doesn't exist
 if [ -f "$observeagent_install_dir/observe-agent.yaml" ]; then
     echo "Leaving existing observe-agent.yaml in place."
 else
     echo "Initializing observe-agent.yaml"
-    sudo $observeagent_install_dir/observe-agent init-config --token $TOKEN --observe_url $OBSERVE_URL --config_path $observeagent_install_dir/observe-agent.yaml
+    INIT_FLAGS="--config_path $observeagent_install_dir/observe-agent.yaml --token $TOKEN --observe_url $OBSERVE_URL --host_monitoring::enabled=true"
+    if [ -n "$LOGS_ENABLED" ]; then
```
Comment (obs-gh-alexlew) [inline scripts/install_mac.sh:60]: where are the keys for these args defined? is this just an env var? are we adding another layer/set of config key names?

Comment (obs-gh-mattcotter) [reply scripts/install_mac.sh:60]: Yes, these are just env vars. I needed some way of getting the args from the connection into the config. Now that you've called attention to this though, I wonder if it's better to switch this to `getopts` to be more in line with the windows script. What do you think about that?

Comment (obs-gh-mattcotter) [reply scripts/install_mac.sh:92]: Updated so it now works like:
`bash scripts/install_mac.sh --token=hi --observe_url example.com --logs_enabled=true`

File: scripts/install_mac.sh [hunk 2]
```diff
@@ -4,9 +4,39 @@ set -e
 
 service_name="com.observeinc.agent"
 observeagent_install_dir="/usr/local/observe-agent"
-# TODO download the file instead of using the local built zip
-script_dir=$(dirname -- "$(readlink -f -- "$0")")
-zip_dir="$(dirname $script_dir)/dist/darwin_arm64_v8.0/observe-agent_Darwin_arm64.zip"
+tmp_dir="/tmp/observe-agent"
+
+# Parse args
+while [ $# -gt 0 ]; do
```
Comment (obs-gh-mattcotter) [inline scripts/install_mac.sh:10]: `getopts` doesn't work on mac for long options unfortunately, so I had to write out this approximation. It doesn't have all the bells and whistles, but is robust enough for our use case of having users copy/paste commands from the connection portal.

**Review by obs-gh-alexlew — APPROVED** [review]

**Pull request merged**